pkg/operator2/deployment.go

```diff
@@ -236,6 +238,24 @@ func getImagePullPolicy(operatorDeployment *appsv1.Deployment) corev1.PullPolicy
 	return containers[0].ImagePullPolicy
 }
 
+func proxyConfigToEnvVars(proxy *configv1.Proxy) []corev1.EnvVar {
```

Comment:
This assumes that the transports that the OAuth server uses honors these proxy headers.  I would like an e2e to prove that.

Reply:
Testing is huge topic that is still unaddressed throughout the whole proxy initiative. This will be follow-up.